app/flatpak-builtins-repo.c

```diff
@@ -205,14 +205,109 @@ print_metadata (GVariant   *meta,
     }
 }
 
+static gboolean
+collect_tree_size (GLnxDirFdIterator *iter,
+                   guint64 *count,
+                   guint64 *actual,
+                   guint64 *total,
+                   GError **error)
+{
+  while (TRUE)
+    {
+      struct dirent *dent;
+
+      if (!glnx_dirfd_iterator_next_dent_ensure_dtype (iter, &dent, NULL, NULL))
+        break;
+
+      if (dent == NULL)
+        break;
+
+      else if (dent->d_type == DT_DIR)
+        {
+          g_auto(GLnxDirFdIterator) iter2 = { 0 };
+
+          if (!glnx_dirfd_iterator_init_at (iter->fd, dent->d_name, FALSE, &iter2, error))
+            return FALSE;
+
+          if (!collect_tree_size (&iter2, count, actual, total, error))
+            return FALSE;
+        }
+      else if (dent->d_type == DT_REG)
+        {
+          struct stat stat;
+
+          if (!glnx_fstatat (iter->fd, dent->d_name, &stat, 0, error))
+            return FALSE;
+
+          if (count)
+            *count += 1;
+          if (actual)
+            *actual += stat.st_size;
+          if (total)
+            *total += stat.st_size * stat.st_nlink;
```

Comment:
As we discussed out of band, this is missleading. A repo with a single checkout would count as being installed twice. Maybe a more realistic value would be `stat.st_size * MAX (1, stat.st_nlink - 1)`. I.e. one deploy is "free".

Reply:
I realized later that the checkout question was a bit misleading, since this command is really only meant to be used on a server-side repo (our local ones usually have neither a summary file nor ostree-metadata, so the repo command won't work on them)

Reply:
and it has now been fixed to give an error message on typical local repos, which won't have a summary file (unless create-usb mistakenly adds one), and only remote ostree-metadata refs.